spf/dnscache.go

```diff
@@ -0,0 +1,26 @@
+package spf
+
+type dnsCache map[string]map[string][]string
```

Comment:
Correct, we could do a similar thing. Our current plan is to support spf flattening in some form, but we are quite opposed to anything in the regular flow that actually depends on DNS lookups. Hence the cache here, that would presumably be updated daily on cron or something like that.